### PR TITLE
Wait for VM to be down before deletion

### PR DESCRIPTION
**Pull request by xiboy**
Hi,

This PR is to fix the error "VM is running" I have when I destroy a VM which is on.

### CODE REVIEW

File: lib/vagrant-ovirt3/action.rb [hunk 1]
```diff
@@ -99,7 +99,9 @@ def self.action_destroy
             end
 
             b2.use ConnectOVirt
-            b3.use ProvisionerCleanup, :before if defined?(ProvisionerCleanup)
+            b2.use HaltVM
+            b2.use WaitTillDown
+            b2.use ProvisionerCleanup, :before if defined?(ProvisionerCleanup)
```
Comment (myoung34) [inline lib/vagrant-ovirt3/action.rb]: this needs to come before `Halt`, it requires access to the VM

File: lib/vagrant-ovirt3/action/create_vm.rb [hunk 2]
```diff
@@ -17,7 +17,7 @@ def call(env)
           config = env[:machine].provider_config
 
           # Gather some info about domain
-          name = (config.name.nil? ? env[:domain_name] : config.name)[0,15]
+          name = (config.name.nil? ? env[:domain_name] : config.name)[0,80]
```
Comment (myoung34) [inline lib/vagrant-ovirt3/action/create_vm.rb]: you can take this out, i'm going to make a specific commit tonight to address this issue

File: lib/vagrant-ovirt3/action/wait_till_down.rb [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+        def initialize(app, env)
+          @logger = Log4r::Logger.new("vagrant_ovirt3::action::wait_till_down")
+          @app = app
+        end
+
+        def call(env)
+
+          env[:ui].info(I18n.t("vagrant_ovirt3.wait_till_down"))
+          for i in 0..20
```
Comment (myoung34) [inline lib/vagrant-ovirt3/action/wait_till_down.rb]: can you use `connect_timeout` similar to `create_vm.rb` line `111` ?

**Comment by myoung34** [comment]
Aside from those 3 changes this looks good.
I'll make a much more elegant solution for the name length tonight.

**Comment by xiboy** [comment]
Ok, done!

File: lib/vagrant-ovirt3/action/create_vm.rb (hunk 2 above)
Comment (myoung34) [inline lib/vagrant-ovirt3/action/create_vm.rb]: can you remove this?

**Comment by myoung34** [comment]
Still just the one lingering change.
Have you tested this on your set up? I wont be able to merge until later tonight to verify.

Best practices: can you squash all your commits into 1? 1 change = 1 commit is my general rule. If not no big deal, just my preference.

**Comment by xiboy** [comment]
Ok, I squash all my commits into 1.

And I revert change for name length, it is not working on my setup, but I have patched it on local copy.

**Pull request merged**